storagenode/console/service.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package console
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/bandwidth"
+	"storj.io/storj/storagenode/pieces"
+)
+
+// Service is handling storage node operator related logic
+type Service struct {
+	log *zap.Logger
+
+	bandwidth bandwidth.DB
+	pieceInfo pieces.DB
+	kademlia  *kademlia.Kademlia
+
+	allocatedBandwidth memory.Size
+	allocatedDiskSpace memory.Size
+	walletNumber       string
+	upTime             time.Time
+}
+
+// NewService returns new instance of Service
+func NewService(log *zap.Logger, bandwidth bandwidth.DB, pieceInfo pieces.DB, kademlia *kademlia.Kademlia,
+	allocatedBandwidth, allocatedDiskSpace memory.Size, walletNumber string) (*Service, error) {
+	if log == nil {
+		return nil, errs.New("log can't be nil")
+	}
+
+	if bandwidth == nil {
+		return nil, errs.New("bandwidth can't be nil")
+	}
+
+	if pieceInfo == nil {
+		return nil, errs.New("pieceInfo can't be nil")
+	}
+
+	if kademlia == nil {
+		return nil, errs.New("kademlia can't be nil")
+	}
+
+	service := Service{
+		log:                log,
+		bandwidth:          bandwidth,
+		pieceInfo:          pieceInfo,
+		kademlia:           kademlia,
+		allocatedBandwidth: allocatedBandwidth,
+		allocatedDiskSpace: allocatedDiskSpace,
+		walletNumber:       walletNumber,
+		upTime:             time.Now(),
+	}
+
+	return &service, nil
+}
+
+// GetUsedBandwidth returns all info about storage node bandwidth usage
+func (s *Service) GetUsedBandwidth(ctx context.Context) (*BandwidthInfo, error) {
+	usage, err := bandwidth.TotalMonthlySummary(ctx, s.bandwidth)
+	if err != nil {
+		return nil, err
+	}
+
+	return FromUsage(usage, s.allocatedBandwidth.Int64())
+}
+
+// GetUsedStorage returns all info about storagenode disk space usage
+func (s *Service) GetUsedStorage(ctx context.Context) (*DiskSpaceInfo, error) {
+	spaceUsed, err := s.pieceInfo.SpaceUsed(ctx)
```

Comment:
`*monitor.Service` can tell you this information.

Reply:
@egonelbre - also storagenode/inspector/inspector.go has such functionality. But, as console/service and monitoring/service they serves for different purposes

Reply:
Inspector in this case is an Endpoint, so only for responding to network requests.